src/app/modules/group/components/group-invite-users/group-invite-users.component.ts

```diff
@@ -1,24 +1,166 @@
+    private messageService: MessageService,
+    ) {}
 
   ngOnInit(): void {}
 
-  onInvite() {
-    this.showMsg = !this.showMsg;
+  private processRequestError(_err: any) {
+    this.messageService.add({
+      severity: 'error',
+      summary: 'Error',
+      detail: ERROR_MESSAGE.fail,
+      life: TOAST_LENGTH,
+    });
   }
 
+  private displayResponse(response: Map<string, InvitationResult>) {
+    const sucessInvites: string[] = [];
+    const alreadyInvited: string[] = [];
+    const notFoundUsers: string[] = [];
+    const invalidInvites: string[] = [];
+
+    for (const [key, value] of response) {
+      switch (value) {
+        case InvitationResult.Success:
+          sucessInvites.push(key);
+          break;
+        case InvitationResult.AlreadyInvited:
+          alreadyInvited.push(key);
+          break;
+        case InvitationResult.Error:
+          invalidInvites.push(key);
+          break;
+        case InvitationResult.NotFound:
+          notFoundUsers.push(key);
+          break;
+      }
+    }
+
+    if (sucessInvites.length > 0)
+      this.messages.push({
+        type: 'success',
+        summary: `${sucessInvites.length} user(s) invited successfully: `,
+        detail: `${sucessInvites.join(', ')}`,
+      });
+
+    if (alreadyInvited.length > 0)
+      this.messages.push({
+        type: 'info',
+        summary: `${alreadyInvited.length} user(s) have already been invited: `,
+        detail: `${alreadyInvited.join(', ')}`,
+      });
+
+    if (notFoundUsers.length > 0)
+      this.messages.push({
+        type: 'error',
+        summary: `${notFoundUsers.length} user login(s) not found: `,
+        detail: `${notFoundUsers.join(', ')}`,
+      });
+
+    if (invalidInvites.length > 0)
+      this.messages.push({
+        type: 'error',
+        summary: `${invalidInvites.length} user login(s) could not be invited: `,
+        detail: `${invalidInvites.join(', ')}`,
+      });
+  }
+
+  /* events */
+
+  changeText(text:string)
+  {
+    this.canClick = true;
+    this.tooManyLogins = false;
+
+    if (text.length == 0)
+    {
+      this.canClick = false;
+      return;
+    }
+
+    this.logins = text.split(',');
+
+    if (this.logins.length >= 100)
+    {
+      this.canClick = false;
+      this.tooManyLogins = true;
+      return;
+    }
+  }
+
+  invite() {
+    // clear the messages
+    this.messages = [];
+
+    // remove empty logins and duplicates
+    const logins = this.logins
+      .map((login) => login.trim())
+      .filter(function (login, index, self) {
+        return self.indexOf(login) == index && login != '';
+      });
+
+    this.logins = [];
+
+    if (logins.length == 0) return;
+
+    // disable UI
+    this.processing = true;
+
+    this.createGroupInvitationsService.createInvitations(this.group.id, logins).subscribe(
+      (res) => {
+        this.displayResponse(res);
+
+        // Clear the textarea
+        if (this.textArea != undefined)
```

Comment:
To test undefined, you can write:
```suggestion
        if (this.textArea)
```